source-control/git-guidance/readme.md

```diff
@@ -1,11 +1,198 @@
 # Git Guidance
 
-## The basics
-
-### Installation
+## Installation
 
 [Install Git](https://git-scm.com/downloads) and follow the [First-Time Git Setup](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup).
 
+## Basic workflow
+
+Whenever you want to make a change to a repository you need to first clone it so you can work on it locally
+
+```cmd
+git clone https://github.com/username/repo-name
+```
+
+To avoid adding code that has not been peer reviewed to the main branch (ex. develop) we typically work in feature branches, and merge these back to the main trunk with a Pull Request
+
+```cmd
+# pull the latest changes
+git pull
+
+# start a new feature branch based on the develop branch
+git chekcout -b feature/123-add-git-instructions develop
+
+# edit some files
+
+git add <file>
+git commit -m "add basic instructions"
+
+# edit some files
+
+git add <file>
+git commit -m "add more advanced instructions"
+
+# push the branch to the remote repository
+git push --set-upstream origin feature/123-add-git-instructions
+```
+
+Once the feature branch is pushed to the remote repository it is visible to anyone with access to the code.
+Now you can open a Pull Request in GitHub, Azure DevOps or similar to request to merge the changes into the main branch.
+
+## Cloning
+
+Cloning a repository pulls down a full copy of all the repository data, so that you can work on it locally.
+This copy includes all versions of every file and folder for the project.
+
+```cmd
+git clone https://github.com/username/repo-name
+```
+
+You only need to clone the repository the first time. Before any subsequent branches you can sync any changes from the remote repository using `git pull`.
+
+## Branching
+
+Often, the `main` or `develop` branch of a repository will be locked so that you can't make changes without a Pull Request.
+In this case it is useful to create a separate branch for your local/feature work, so that you can work and track your changes without modifying the main trunk.
+
+Pull the latest changes and create a new branch for your work based on the trunk (in this case develop)
+
+```cmd
+git pull
+git checkout -b feature/feature-name develop
+```
+
+At any point, you can move between the branches with `git checkout <branch>` as long as you have committed or stashed your work.
+If you forget the name of your branch use `git branch --all` to list all branches
+
+When you are done working, push your changes to a branch in the remote repository using `git push`
+
+```cmd
+git push --set-upstream origin feature/feature-name
```

Comment:
I would add an example of a regular 'git push' (once the branch is already linked to a remote branch) here too